exatomic/core/universe.py

```diff
@@ -349,3 +349,59 @@ def basis_function_contributions(universe, mo, mocoefs='coef',
         return joined
     else:
         raise NotImplementedError("not clever enough for that.")
+
+
+def to_cube(universe,file_name='output',field_number=0)
```

Comment:
Would it make more sense to put this function in `exatomic.interfaces.cube`? Specifically, would it make sense to add this functionality as a method on `exatomic.interfaces.cube.Cube`? If you look [here](https://github.com/exa-analytics/exatomic/blob/edcbc8b5198f8b86185741fa1aa0404b2836b516/exatomic/interfaces/cube.py#L94) there is already some work to pull a cube from universe fields.

Reply:
Agreed, I believe some of the functionality this PR addresses may already be implemented elsewhere in the code.

```python
import exatomic
uni = exatomic.Universe.load(exatomic.base.resource('mol-carbon-dz-valid.hdf5'))
cube = exatomic.Cube.from_universe(uni, 0)
cube.write('output.cube')
```

I think the `to_cube` method fits better as a convenience method on the `Universe` object that recycles the `exatomic.interfaces.cube.Cube` API.

Reply:
> Agreed, I believe some of the functionality this PR addresses may already be implemented elsewhere in the code.
> 
> ```python
> import exatomic
> uni = exatomic.Universe.load(exatomic.base.resource('mol-carbon-dz-valid.hdf5'))
> cube = exatomic.Cube.from_universe(uni, 0)
> cube.write('output.cube')
> ```
> 
> I think the `to_cube` method fits better as a convenience method on the `Universe` object that recycles the `exatomic.interfaces.cube.Cube` API.

Hmmm... I just tested this on my cases, the outputs are the same to `to_cube` function.
Happy to know this functionality already exists, it might be better to discard this PR just not to mess up the code architecture.

Reply:
> I think the `to_cube` method fits better as a convenience method on the `Universe` object that recycles the `exatomic.interfaces.cube.Cube` API.

Or as Tom has said, put the `to_cube` method as a convenience method on the `Universe` object, would it be better to define it as the `Universe` class function or just in the `universe.py` as it is?

Reply:
Yeah I would not discard this PR. It has an important bugfix. I think it makes sense to leverage existing functionality and keep things DRY (don't repeat your/ourselves). A method on the Universe class that leverages interfaces.cube.Cube.from_universe may be the best approach.

Reply:
Right, the bugfix is greatly appreciated and I think that since you had to go through the work of your own implementation, that indicates to me that the available API is not obvious or convenient.

I think it makes a lot of sense to have the `to_cube` method on the `Universe` directly which may make it more apparent that the functionality exists.

It should not be too much work to update this PR to reflect those changes. Consider it accepted with major revisions :)

Reply:
Yeah!~~~

Reply:
Oh, wait, let me replace my codes with `exatomic.interfaces.cube.Cube` API.